pkg/validation/validate/ratcheting.go

```diff
@@ -0,0 +1,107 @@
+package validate
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"k8s.io/kube-openapi/pkg/validation/strfmt"
+)
+
+type ratchetingSchemaValidator struct {
+	Schema       *spec.Schema
+	Root         interface{}
+	Path         string
+	KnownFormats strfmt.Registry
+	Options      []Option
+}
+
+func NewRatchetingSchemaValidator(schema *spec.Schema, rootSchema interface{}, root string, formats strfmt.Registry, options ...Option) *ratchetingSchemaValidator {
+	return &ratchetingSchemaValidator{
+		Schema:       schema,
+		Root:         rootSchema,
+		Path:         root,
+		KnownFormats: formats,
+		Options:      options,
+	}
+}
+
+func (r *ratchetingSchemaValidator) ValidateUpdate(old, new interface{}) *Result {
+	opts := append([]Option{
+		r.enableRatchetingOption(old),
+	}, r.Options...)
+
+	s := NewSchemaValidator(r.Schema, r.Root, r.Path, r.KnownFormats, opts...)
+
+	res := s.Validate(new)
+
+	if res.IsValid() {
+		return res
+	}
+
+	if reflect.DeepEqual(old, new) {
+		//!TODO: only consider errors with paths on "."
```

Comment:
yes, without filtering by `.` we will deepequal all the way up to the route. That's heavy and unnecessary.